### PR TITLE
SDIT-1978: Suppress if-match DoS and upgrade kotlin and other libs

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -31,7 +30,7 @@ fun isNonStable(version: String): Boolean {
 }
 
 group = "uk.gov.justice.hmpps.gradle"
-version = "6.0.2"
+version = "6.0.3"
```
Comment (petergphillips) [inline build.gradle.kts:33]: no beta here 😁

File: build.gradle.kts [hunk 2]
```diff
@@ -62,19 +61,12 @@ dependencies {
...
 
-  testImplementation("org.junit.jupiter:junit-jupiter:5.10.3")
+  testImplementation("org.junit.jupiter:junit-jupiter:5.11.0")
   testImplementation("org.mockito:mockito-junit-jupiter:5.12.0")
   testImplementation("org.assertj:assertj-core:3.26.3")
   testImplementation("net.javacrumbs.json-unit:json-unit-assertj:3.4.1")
   testImplementation("com.google.code.gson:gson:2.11.0")
   testImplementation("org.eclipse.jgit:org.eclipse.jgit:6.10.0.202406032230-r")
-  // Had to include this when I had the same error as https://youtrack.jetbrains.com/issue/KT-49547, this links to https://github.com/gradle/gradle/issues/16774 which has includes a workaround
-  testRuntimeOnly(
```
Comment (petergphillips) [inline build.gradle.kts:72]: `serviceOf` got errors with gradle 8.10 so tried taking it out and it worked

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/AppInsightsConfigManager.kt [hunk 3]
```diff
@@ -4,8 +4,8 @@ import org.gradle.api.Project
 import org.gradle.api.tasks.Copy
 import uk.gov.justice.digital.hmpps.gradle.ConfigManager
 
-private const val APP_INSIGHTS_VERSION = "3.5.3"
-const val OPENTELEMETRY_VERSION = "1.38.0"
+private const val APP_INSIGHTS_VERSION = "3.5.4"
+const val OPENTELEMETRY_VERSION = "1.40.0"
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/AppInsightsConfigManager.kt:8]: this matches the version released by 3.5.4 - see https://github.com/microsoft/ApplicationInsights-Java/blob/4779ed8eb97b1bffc058cc765c6ab3e4d8063f4b/dependencyManagement/build.gradle.kts

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**